### ISSUE TITLE
Unexpected anchor behaviour when using functions

### ISSUE THREAD

**Issue by groundnuty** [k14s/ytt#481]
**What steps did you take:**

Input:
```
#@ def group_alpha():
name: alpha
#@ end
groups: &all_groups
  - #@ group_alpha()
foo:
  groups: *all_groups
```

Result:
```
groups:
- null
foo:
  groups:
  - name: alpha
```

Expected: 
```
groups:
- alpha
foo:
  groups:
  - name: alpha
```

Expected according to [limitations](https://carvel.dev/ytt/docs/latest/known-limitations/)
```
groups:
- alpha
foo:
  groups:
  - null
```

[Interactive](https://carvel.dev/ytt/#gist:https://gist.github.com/groundnuty/863b0410116a67b4232f6cd318d10fde) reproduction from gist.

**Anything else you would like to add:**
Please explain this unexpected behaviour, as it's not consistent with #172 or #186 

**Environment:**

- ytt version 0.36.0
- macos 10.14 (18A391)

**Comment by cppforlife** [comment]
hey @groundnuty, as you found this is one of the known limitations (https://carvel.dev/ytt/docs/latest/known-limitations/); however, we should have probably made more clear that results are "undefined" (instead of providing _some_ example). currently (and that may change) last node that splats the anchor would receive data. for your example:

```
#@ def group_alpha():
name: alpha
#@ end
groups: &all_groups
  - #@ group_alpha()
foo:
  groups: *all_groups  <--- this will be nil
  foo: *all_groups  <--- this will actually contain it
```

tldr -- we should improve docs to clarify that you should not rely on this behaviour.

**Comment by Rajdeep1311** [comment]
I want to get involved in this project can anybody please help? how to get started?
